src/components/ProgramForm/ThreeColProgram/List/index.tsx

```diff
@@ -25,74 +21,15 @@ export const CurriculumList = () => {
     }
   }, [addNewLesson, state]);
 
-  /*
-  const onSort = useCallback(
-    (lesson: API.Lesson, up: boolean) => {
-      return lesson.id && sortLesson && sortLesson(lesson.id, up);
-    },
-    [sortLesson],
-  );
-  */
-
-  const addNewTopicToLesson = useCallback(
-    (lesson_id: number, topic_type: TopicType) => {
-      if (addNewTopic && state) {
-        const newTopic = addNewTopic(lesson_id, topic_type);
-        history.push(`/courses/list/${state.id}/program/?topic=${newTopic.id}`);
-      }
-    },
-    [addNewTopic, state],
-  );
-
   if (state && state.lessons) {
     return (
       <aside
         className={`program-sidebar program-sidebar--left ${
           state.lessons.length > 0 ? '' : 'program-sidebar--empty'
         }`}
       >
-        {state.lessons.length > 0 ? (
-          <ul className={'program-sidebar__list'}>
-            {state.lessons
-              .sort((lessonA, lessonB) => (lessonA.order || 0) - (lessonB.order || 0))
-              .map((lesson) => (
-                <li key={lesson.id}>
-                  <div className={'program-sidebar__list-item program-sidebar__list-item--lesson'}>
-                    <NavLink to={`/courses/list/${state.id}/program/?lesson=${lesson.id}`}>
-                      <FolderOutlined />
-                      {lesson.title}
-                    </NavLink>
-
-                    <TopicTypesSelector
-                      onSelected={(topic_type) =>
-                        lesson.id && addNewTopicToLesson(lesson.id, topic_type)
-                      }
-                    />
-                  </div>
-
-                  {lesson.topics && lesson.topics?.length > 0 && (
-                    <ul>
-                      {lesson.topics?.map((topic) => (
-                        <li
-                          key={topic.id}
-                          className={
-                            currentEditMode &&
-                            currentEditMode.mode === 'topic' &&
-                            currentEditMode.id === topic.id
-                              ? 'program-sidebar__list-item program-sidebar__list-item--active'
-                              : 'program-sidebar__list-item'
-                          }
-                        >
-                          <NavLink to={`/courses/list/${state.id}/program/?topic=${topic.id}`}>
-                            {getTypeIcon(getTypeName(topic))} {topic.title}
-                          </NavLink>
-                        </li>
-                      ))}
-                    </ul>
-                  )}
-                </li>
-              ))}
-          </ul>
+        {state.id && state.lessons.length > 0 ? (
+          <LessonList courseId={state.id} courseLessons={state.lessons} />
         ) : (
           <Empty />
         )}
```

Comment:
```suggestion
          <LessonList  />    
```

you are using const { addNewTopic, currentEditMode } = useContext(Context); inside this component so what's the point of passing some values from HOC istead of fetching all or nothing from context